File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) { 
        
       int profit = 0;
       int buyDay = 0;
        
       for(int sellDay=1; sellDay<prices.length; sellDay++){
           if(prices[buyDay] >= prices[sellDay]){ 
               buyDay = sellDay;
           }
            profit = Math.max(profit,prices[sellDay] - prices[buyDay]);
       }
        
        return profit;
   
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let minPrice = Number.MAX_VALUE;
    let maxProfit = 0;

    for(let price of prices){
        minPrice = Math.min(minPrice,price);
        let currentPrice = price - minPrice;
        maxProfit = Math.max(maxProfit,currentPrice);
    }

    return maxProfit;
};